src/components/cesium_map/cesium_UI.js

```diff
@@ -0,0 +1,148 @@
+
+
+let viewer = null;
+let bbox = null;
+let setPoints = null;
+let setPrimitive = null;
+let searchFields = null;
+const api = new ISamplesAPI();
+const moorea = new SpatialView(-149.8169236266867, -17.451466233002286, 2004.7347996772614, 201.84408760864753, -20.853642866175978);
+const patagonia = new SpatialView(-69.60169132023925, -54.315990127766646, 1781.4560051617016, 173.54573250470798, -15.85292472305027);
+const Global = new SpatialView(-149.8169236266867, -17.451466233002286, 15000000, 90.0, -90);
+
+async function countRecordsInBB(bb) {
+  const Q = bb.asSolrQuery('producedBy_samplingSite_location_rpt');
+  return await api.countRecordsQuery({ Q: Q, searchFields: searchFields, rows: 0 });
+}
+
+function showCoordinates(lon, lat, height) {
+  let e = document.getElementById("position");
+  e.innerText = `${lat.toFixed(4)}, ${lon.toFixed(4)}, ${height.toFixed(1)}`;
+}
+
+function clearBoundingBox() {
+  viewer.removeEntity(bbox);
+  setPoints.clear();
+  document.getElementById("clear-bb").style.display = "none";
+}
+
+async function selectedBoxCallbox(bb) {
+  let text = `Record count : ${await countRecordsInBB(bb)}`;
+
+  viewer.removeEntity(bbox);
+  bbox = viewer.addRectangle(bb, text)
+
+  const Q = bb.asSolrQuery('producedBy_samplingSite_location_rpt');
+  setPoints.loadApi({ Q: Q, searchFields: searchFields, rows: 5000 })
+
+  const btn = document.getElementById("clear-bb");
+  btn.style.display = "block";
+  btn.onclick = clearBoundingBox;
+}
+
+class CesiumMap extends React.Component {
+
+  // https://reactjs.org/docs/refs-and-the-dom.html
+  // Use react ref to manipulate DOM directly
+  // constructor(props) {
+  //   super(props);
+  // }
+
+  // This is a initial function in react liftcycle.
+  // Only call once when this component first render
+  componentDidMount() {
+    searchFields = this.props.searchFields;
+    viewer = new ISamplesSpatial("cesiumContainer");
+    // viewer.addPointsBySource(642092);
+    viewer.visit(moorea);
+    viewer.addHud("cesiumContainer");
+    viewer.trackMouseCoordinates(showCoordinates);
+    viewer.enableTracking(selectedBoxCallbox);
+    setPrimitive = new PointStreamPrimitiveCollection("Primitive Points");
+    viewer.addPointPrimitives(setPrimitive)
+    viewer.addDataSource(new PointStreamDatasource("BB points")).then((res) => { setPoints = res });
+    setPrimitive.load({ searchFields: this.props.searchFields, rows: 50000 });
+  }
+
+  // https://medium.com/@garrettmac/reactjs-how-to-safely-manipulate-the-dom-when-reactjs-cant-the-right-way-8a20928e8a6
+  // manipulate Dom outside the react model
+  shouldComponentUpdate(nextProps) {
+    const key1 = JSON.stringify(nextProps.searchFields)
+    const key2 = JSON.stringify(this.props.searchFields)
+    if (key1 !== key2) {
```

Comment:
Convert the state of the field list to string and check if the new state changes.